src/sinol_make/commands/export/__init__.py

```diff
@@ -92,11 +92,16 @@ def create_makefile_in(self, target_dir: str, config: dict):
         with open(os.path.join(target_dir, 'makefile.in'), 'w') as f:
             cxx_flags = '-std=c++17'
             c_flags = '-std=c17'
+            def to_array(obj):
+                if isinstance(obj, str):
+                    return [obj]
+                return obj
+
             if 'extra_compilation_args' in config:
                 if 'cpp' in config['extra_compilation_args']:
-                    cxx_flags += ' ' + ' '.join(config['extra_compilation_args']['cpp'])
+                    cxx_flags += ' ' + ' '.join(to_array(config['extra_compilation_args']['cpp']))
                 if 'c' in config['extra_compilation_args']:
-                    c_flags += ' ' + ' '.join(config['extra_compilation_args']['c'])
+                    c_flags += ' ' + ' '.join(to_array(config['extra_compilation_args']['c']))
```

Comment:
The `to_array` is a *very specific* function. I bet we can make a function whose purpose is much clearer:

```
def format_multiple_arguments(obj):
  if isinstance(obj, str):
    return str
  return ' '.join(obj)

cxx_flags += ' ' + format_multiple_arguments(config[...])
```
Shorter and more readable!

Maybe before you put your code for review, could you look over it and think *can this piece of code be made shorter?*

Reply:
I changed it